theano/compile/function_module.py

```diff
@@ -280,7 +280,8 @@ class Function(object):
     """
 
     def __init__(self, fn, input_storage, output_storage, indices, outputs,
-                 defaults, unpack_single, return_none, maker):
+                 defaults, unpack_single, return_none, output_dictionary_flag, 
+                 output_keys, maker):
```

Comment:
Might be safe here, but in general try adding new arguments as last arguments, to be sure not to break any existing code.